target/linux/ath79/files/drivers/mtd/nand/rb4xx_nand.c

```diff
@@ -0,0 +1,399 @@
+/*
```

Comment:
This is a direct copy of https://github.com/openwrt/openwrt/blob/master/target/linux/ar71xx/files/drivers/mtd/nand/rb4xx_nand.c with a few changes (annotated below)